### DISCUSSION TITLE
Surprising aspects of v0.3 CLI

### DISCUSSION THREAD

**Discussion by dbr** [taskchampion/taskchampion#153] [Q&A] [answered]
The CLI is a bit surprising, possibly in part because I've not really used taskwarrior:

1. If I add some tasks, then run "task start 1" it clobbers all task descriptions with the string "1".  Same with "task start --help".
239.21

I think taskwarrior supports both `task 1 start` and `task start 1`. It prints a "are you sure?" message on `task start`.

2. Having batch operations highlights need for at least a basic undo system

3. I'd "almost" suggest we require quoting the task text, as it removes ambiguity in behaviour
 
For example 'task add --help" seems like a reasonable thing to try, but not clear how to support both that producing help text, and "task add Implement --help for my app" adding a new task. Whereas if tasks strings need quoted, like `task add "Implement --help for my app" ` the behaviour is is well established

I'm not sure about this as it breaks from taskwarrior in quite an obvious way

**Comment by djmitche** [comment, marked as answer]
Let's get an issue filed for `task start 1` -- that should work.  In general, I think things are almost always unambiguous, and with sufficient testing we can ensure we do the "natural" thing in the corner cases.

Similarly, I think it makes sense to reproduce some of TW's "are you sure" prompts -- those were good UX decisions!  I think at least one of them is based on the number of tasks affected?

Let's get an issue filed to add `--help` support for each subcommand, too.  That's an exception to the CLI model that doesn't typically use `--xyz` options, because people do tend to guess `--help`!

Undo is #92.  It's hard!

Quoting is always *available*, but I think we can continue to get away without requiring it.  So `task add "Implement --help for my app"` already works.  The escape hatch for things like creating a task with description "--help" (for which quoting won't be of any use) is `task add -- --help`.  All of that could certainly use some better documentation.

**Reply by djmitche** [reply]
@dbr did you want to add some issues for the above, and mark this discussion as answered?

**Reply by dbr** [reply]
Yep done!

https://github.com/djmitche/taskchampion/issues/177 - `task start 1`
https://github.com/djmitche/taskchampion/issues/178 - allowing help arg
https://github.com/djmitche/taskchampion/issues/179 - "Are you sure" prompts

**Reply by dbr** [reply]
Hmm, it's only possible to mark "Q&A" threads as completed (which I changed this to be as a test, but.. it's a bit clunky)
